### DISCUSSION TITLE
Future values for multivariate predictions

### DISCUSSION THREAD

**Discussion by KajetanPoliak** [winedarksea/AutoTS#179] [Q&A]
Hi everybody,

 I have a dataset with 20+ features and trying to predict one of them 6 months into the future. Some of these features have known values for these 6 months to the future, how can I help the training/forecasting with these values. I have tried using `create_regressor` but am unsure how I can add specific feature values. 
Do you have any recommendations/ideas?

Thank you very much.

K.

**Comment by winedarksea** [comment]
so what create_regressor does it is shift values that aren't known from the future into the future so they work as a `future_regressor`
what I would do is use create_regressor on all series for which you won't know the future values, giving you future_regressor_train_a and future_regressor_fcst_a, then concat these with future_regressor_train/fcst_b, where this b group is your feature set for which you already know the values. Make sense?

**Reply by KajetanPoliak** [reply]
Thank you for your fast and great answer. This seems like the approach. What I don't understand is how can I specify multiple features in the `future_regressor`. When I create one using 'create_regressor' I get 10 lagged values for each timestamp - I dont know what feature is being lagged. I can always send there a series with just one feature then I will get 10 lagged values of known feature. When I do this for all features where I dont know the future + I will create future_regressor for known features how will the model know which columns correspond to which feature?
Maybe I am off with understanding future_regressor. Does my question make sense?
Thank you!

**Reply by winedarksea** [reply]
Firstly, you should know that `future_regressor` is optional. Internally, for machine learning models (ie MultivariateRegression with regression models like LightGBM and ExtraTrees) it will automatically create a feature sets based on parameters. These lagged features you are creating here are more for models like ARIMA which can accept a regressor but won't have a _multivariate_ lag unless this future regressor is passed.

Try passing `summarize=None` to `create_regressor`. By default it lags all features and then aggregates, down to 10, to keep things faster and cleaner. Removing the summarization will make it more clear what is being done.

I don't quite understand what you are asking with ` how will the model know which columns correspond to which feature`. Column names must match between the training and forecast sets of the `future_regressor`. As for how those are used by the model, well, the model learns weights based off what is most useful to prediction, so it figures out that form of correspondence on its own.

**Reply by KajetanPoliak** [reply]
Great! Removing the summarization made it all clear. I will continue with this approach then. :) Thank you and have a great day. Your help is much appreciated.
